packages/terra-icon/src/IconBase.jsx

```diff
@@ -50,41 +50,68 @@ const defaultProps = {
   focusable: 'false',
 };
 
-const IconBase = ({
-  isBidi,
-  isSpin,
-  children,
-  height,
-  width,
-  ariaLabel,
-  focusable,
-  ...customProps
-}) => {
-  const attributes = Object.assign({}, customProps);
+class IconBase extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addAnimationStyles();
+  }
 
-  // append to existing classNames
-  const classes = cx(
-    'icon',
-    { 'is-bidi': isBidi },
-    { 'is-spin': isSpin },
-    attributes.className,
-  );
+  addAnimationStyles() {
+    const { isSpin } = this.props;
 
-  // aria-label is present, remove aria-hidden, set role to img
-  if (ariaLabel) {
-    attributes['aria-label'] = ariaLabel;
-    attributes.role = 'img';
-    attributes['aria-hidden'] = null;
-  } else {
-    attributes['aria-hidden'] = 'true';
+    if (isSpin) {
+      const css = '@-webkit-keyframes tui-spin-ltr{0%{-webkit-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(359deg);transform:rotate(359deg)}}@keyframes tui-spin-ltr{0%{-webkit-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(359deg);transform:rotate(359deg)}}@-webkit-keyframes tui-spin-rtl{0%{-webkit-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(-359deg);transform:rotate(-359deg)}}@keyframes tui-spin-rtl{0%{-webkit-transform:rotate(0deg);transform:rotate(0deg)}100%{-webkit-transform:rotate(-359deg);transform:rotate(-359deg)}}[dir=ltr] .tui-Icon.is-spin {-webkit-animation: tui-spin-ltr 2s infinite linear;animation: tui-spin-ltr 2s infinite linear;}[dir=rtl] .tui-Icon.is-spin {-webkit-animation: tui-spin-rtl 2s infinite linear;animation: tui-spin-rtl 2s infinite linear;}';
```

Comment:
This output is what we see running the styles through autoprefixer currently. I've named spaced the animation name, `tui-spin-ltr` and `tui-spin-rtl` as well as minified the CSS as much as reasonable since we'll write this string inline to the DOM.